pymc3/distributions/bound.py

```diff
@@ -14,196 +14,97 @@
 
 from numbers import Real
 
-import aesara.tensor as at
 import numpy as np
 
+from aesara.tensor.random.op import RandomVariable
+from aesara.tensor.var import TensorVariable
+
 from pymc3.aesaraf import floatX
-from pymc3.distributions import transforms
+from pymc3.distributions import _logp
+from pymc3.distributions.continuous import BoundedContinuous
 from pymc3.distributions.dist_math import bound
-from pymc3.distributions.distribution import Continuous, Discrete, Distribution
+from pymc3.distributions.distribution import Continuous, Discrete
+from pymc3.model import modelcontext
 
 __all__ = ["Bound"]
 
 
-class _Bounded(Distribution):
-    def __init__(self, distribution, lower, upper, default, *args, **kwargs):
-        self.lower = lower
-        self.upper = upper
-        self._wrapped = distribution.dist(*args, **kwargs)
+class BoundRV(RandomVariable):
+    name = "bound"
+    ndim_supp = 0
+    ndims_params = [0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("Bound", "\\operatorname{Bound}")
 
-        if default is None:
-            defaults = self._wrapped.defaults
-            for name in defaults:
-                setattr(self, name, getattr(self._wrapped, name))
-        else:
-            defaults = ("_default",)
-            self._default = default
-
-        super().__init__(
-            shape=self._wrapped.shape,
-            dtype=self._wrapped.dtype,
-            initval=self._wrapped.initval,
-            defaults=defaults,
-            transform=self._wrapped.transform,
-        )
-
-    def logp(self, value):
+    @classmethod
+    def rng_fn(cls, rng, distribution, lower, upper, size):
+        raise NotImplementedError("Cannot sample from a bounded variable")
+
+
+boundrv = BoundRV()
+
+
+class _DiscreteBounded(Discrete):
+    rv_op = boundrv
+
+    def __new__(cls, *args, **kwargs):
+        transform = kwargs.get("transform", None)
+        if transform is not None:
+            raise ValueError("Cannot transform discrete variable.")
```

Comment:
Codecov is suggesting some important error branches are not covered by the new tests